emcc.py

```diff
@@ -1441,10 +1441,17 @@ def default_setting(name, new_default):
 
   # Wasm SjLj cannot be used with Emscripten EH
   if settings.SUPPORT_LONGJMP == 'wasm':
+    # DISABLE_EXCEPTION_THROWING is 0 by default for Emscripten EH throwing, but
+    # Wasm SjLj cannot be used with Emscripten EH. We error out if
+    # DISABLE_EXCEPTION_THROWING=0 is explicitly requested by the user;
+    # otherwise we disable it here.
+    if 'DISABLE_EXCEPTION_THROWING' in settings_map and settings_map['DISABLE_EXCEPTION_THROWING'] == '0':
+      exit_with_error('SUPPORT_LONGJMP=wasm cannot be used with DISABLE_EXCEPTION_CATCHING=0')
+    settings.DISABLE_EXCEPTION_THROWING = 1
```

Comment:
Can you can use `default_setting('DISABLE_EXCEPTION_THROWING', 1)` to change the default.. and then just check the value. e.g.

```
default_setting('DISABLE_EXCEPTION_THROWING', 1)
if settings.DISABLE_EXCEPTION_THROWING:
   exit_with_error(...)
```

Reply:
Not sure what you mean. Doesn't `default_setting` set `settings.DISABLE_EXCEPTION_THROWING` to 1? Then wouldn't the next `if` is always taken and we exit with the error?

Also, what do you think about @kripken's idea of using `settings_map` instead? I changed the code to use that.

Reply:
No `default_setting` would only set the setting if the user did not yet it.  Do `if not settings.DISABLE_EXCEPTION_THROWING:` right after `default_setting` is that way to detect that the user explicitly disabled it.

Reply:
The `default_setting` function is bases on `settings_map` under the hood.  I think the preferred pattern in this file is to use `default_setting` rather than explicitly checking `settings_map` before unconditionally assigning to a setting.

Reply:
Thanks! Done.

Reply:
> Can you can use `default_setting('DISABLE_EXCEPTION_THROWING', 1)` to change the default.. and then just check the value. e.g.
> 
> ```
> default_setting('DISABLE_EXCEPTION_THROWING', 1)
> if settings.DISABLE_EXCEPTION_THROWING:
>    exit_with_error(...)
> ```

And I guess you meant `if not settings.DISABLE_EXCEPTION_THROWING:` right? (added `not`)

Reply:
Yes, sorry.